src/ngraph/frontend/onnx_import/utils/reduction.cpp

```diff
@@ -32,17 +32,23 @@ namespace ngraph
             {
                 AxisSet get_reduction_axes(const Node& node)
                 {
-                    auto reduction_axes =
-                        node.get_attribute_value<std::vector<std::int64_t>>("axes", {});
-                    std::vector<std::size_t> normalized_axes =
+                    const auto data_ps = node.get_ng_inputs().at(0)->get_output_partial_shape(0);
+                    NGRAPH_CHECK(
+                        data_ps.rank().is_static(),
+                        "Input's rank is required to be static to be able to normalize the axes");
+
+                    const auto reduction_axes =
+                        node.get_attribute_value<std::vector<int64_t>>("axes", {});
+
+                    std::vector<size_t> normalized_axes =
                         ngraph::normalize_axes(node.get_description(),
                                                reduction_axes,
-                                               node.get_ng_inputs().at(0)->get_shape().size());
+                                               static_cast<int64_t>(data_ps.rank()));
 
                     if (reduction_axes.empty())
                     {
-                        normalized_axes = onnx_import::common::get_monotonic_range<std::size_t>(
-                            node.get_ng_inputs().at(0)->get_shape().size());
+                        normalized_axes = onnx_import::common::get_monotonic_range<size_t>(
```

Comment:
You may need to check for static rank inside this `if`

Reply:
It's already checked in line 37

Reply:
I've added this check after the one from line 37 has been removed.